LayoutKitTests/TextViewLayoutTests.swift

```diff
@@ -0,0 +1,152 @@
+            string: textString,
+            attributes: [NSFontAttributeName: UIFont.helvetica(size: 15)])
+        let attributedString2 = NSMutableAttributedString(
+            string: textString,
+            attributes: [NSFontAttributeName: UIFont.helvetica(size: 12)])
+        attributedString1.append(attributedString2)
+        let attributedText = Text.attributed(attributedString1)
+
+        let textView = UITextView(text: attributedText)
+
+        let layout = TextViewLayout(text: attributedText)
+        let layoutView = layout.arrangement()
+
+        XCTAssertEqual(layoutView.frame.size, textView.intrinsicContentSize)
+    }
+
+    func testLineFragmentPadding() {
+        let textString = "Hello World\nHello World\nHello World\nHello World\nHello World"
+        let text = Text.unattributed(textString)
+        let lineFragmentPadding: CGFloat = 30.0
+
+        let textView = UITextView(text: text,
+                                  lineFragmentPadding: lineFragmentPadding)
+
+        let layout = TextViewLayout(text: text, lineFragmentPadding: lineFragmentPadding)
+        let layoutView = layout.arrangement()
+
+        XCTAssertEqual(layoutView.frame.size, textView.intrinsicContentSize)
+    }
+
+    func testTextContainerInset() {
+        let textString = "Hello World\nHello World\nHello World\nHello World\nHello World"
+        let text = Text.unattributed(textString)
+        let textContainerInset = UIEdgeInsets(top: 2, left: 3, bottom: 4, right: 5)
+
+        let textView = UITextView(text: text,
+                                  textContainerInset: textContainerInset)
+
+        let layout = TextViewLayout(text: text, textContainerInset: textContainerInset)
+        let layoutView = layout.arrangement()
+
+        XCTAssertEqual(layoutView.frame.size, textView.intrinsicContentSize)
+    }
+
+}
+
+// MARK: - private helper extension
+
+private extension UITextView {
+
+    static let defaultFont = UIFont.systemFont(ofSize: 12)
+
+    convenience init(text: Text,
+                     font: UIFont? = nil,
+                     lineFragmentPadding: CGFloat = 0,
+                     textContainerInset: UIEdgeInsets = UIEdgeInsets.zero,
+                     contentInset: UIEdgeInsets = UIEdgeInsets.zero,
+                     layoutMargins: UIEdgeInsets = UIEdgeInsets.zero) {
+        self.init()
+
+        switch text {
+        case .unattributed(let unattributedText):
+            self.text = unattributedText
+            self.font = font ?? UITextView.defaultFont
+        case .attributed(let attributedText):
+            // If font is using default, applied default font to attributedString
+            self.attributedText = attributedText.with(defaultFont: font ?? UITextView.defaultFont)
+        }
+
+        self.textContainer.lineFragmentPadding = lineFragmentPadding
+        self.textContainerInset = textContainerInset
+        self.contentInset = contentInset
+        self.layoutMargins = layoutMargins
```

Comment:
As discussed, `contentInset` is used before iOS 7, which we are not supporting. `LayoutMargins` uses for auto layout related.